### ISSUE TITLE
Record fields produce "undeclared value" errors after declaration 

### ISSUE THREAD

**Issue by alexandra-hanson** [melt-umn/Oberon0#4]
Declaring a record type and its fields does not actually declare the fields for usage. For example, from the record module in the positive L4 examples (on the add-examples branch):

```
TYPE RecordTest = RECORD
       foo: INTEGER;
       bar: BOOLEAN
     END;

VAR r: RecordTest;

BEGIN
  r.foo := 42;
  r.bar := TRUE;
  Write(r.foo); WriteLn
```

Results in the following errors:

```
error: Undeclared value foo
error: Undeclared value bar
error: Undeclared value foo
```

This should not be the case.

**Comment by alexandra-hanson** [comment]
I think the issue is in the fieldAccess production in grammars/edu.umn.cs.melt.Oberon0/constructs/dataStructures/abstractSyntax/Expr.sv:

```
abstract production fieldAccess
e::LExpr ::= rec::LExpr fld::Name
{
  e.pp = pp:ppConcat([rec.pp, pp:text("."), fld.pp]);
  
  propagate errors;  --T2
  e.evalConstInt = nothing();  --T2
}
```
I don't think the `fld` has the context (or scope?) it needs from the `rec` in order to find the field when `Name` calls `n.lookupName = lookupValue(s, n.env);`. I think that we want to give the `env` of `rec` to `fld`... trying to recall how I might do this and determine if this is actually what the problem is.

**Comment by ericvanwyk** [comment]
This was trickier than I expected. The LDTA language challenge required splitting the specification into grammars that added new language constructs and new analyses, so equations and aspects are spread over multiple directories, making it hard to follow.  Furthermore, tasks T2 (name analysis) and T3 (type checking) cannot be so cleanly separated for records since we need to know the type of a record in order to access fields on it.

The change was to provide the proper environment to the field in a record access, so that the field, of type `Name`, did not generate any spurious error messages.  The error checking is done on the record access aspect production, the comment there explains other possible solutions.

@alexandra-hanson can you check that all the other tests pass and, if they do, then close this and merge this into develop.  We may want to do another release at some point, but let's wait to see if the LSP work involves more changes and release things then.
